cpp/test/sg/rf_treelite_test.cu

```diff
@@ -0,0 +1,403 @@
+      params.n_inference_rows, Compare<float>(), stream));
+  }
+
+  void SetUp() override {
+    params = ::testing::TestWithParam<RfInputs<T>>::GetParam();
+
+    DecisionTree::DecisionTreeParams tree_params;
+    set_tree_params(tree_params, params.max_depth, params.max_leaves,
+                    params.max_features, params.n_bins, params.split_algo,
+                    params.min_rows_per_node, params.bootstrap_features,
+                    params.split_criterion, false);
+    set_all_rf_params(rf_params, params.n_trees, params.bootstrap,
+                      params.rows_sample, tree_params);
+    // print(rf_params);
+
+    data_len = params.n_rows * params.n_cols;
+    inference_data_len = params.n_inference_rows * params.n_cols;
+
+    allocate(data_d, data_len);
+    allocate(inference_data_d, inference_data_len);
+
+    allocate(labels_d, params.n_rows);
+    allocate(predicted_labels_d, params.n_inference_rows);
+
+    treelite_predicted_labels.resize(params.n_inference_rows);
+    ref_predicted_labels.resize(params.n_inference_rows);
+
+    CUDA_CHECK(cudaStreamCreate(&stream));
+    handle.setStream(stream);
+
+    forest = new typename ML::RandomForestMetaData<T, L>;
+    null_trees_ptr(forest);
+  }
+
+  void TearDown() override {
+    CUDA_CHECK(cudaStreamDestroy(stream));
+
+    CUDA_CHECK(cudaFree(data_d));
+    CUDA_CHECK(cudaFree(inference_data_d));
+    CUDA_CHECK(cudaFree(labels_d));
+    CUDA_CHECK(cudaFree(predicted_labels_d));
+
+    delete[] forest->trees;
+    delete forest;
+    labels_h.clear();
+    predicted_labels_h.clear();
+    data_h.clear();
+    inference_data_h.clear();
+    treelite_predicted_labels.clear();
+    ref_predicted_labels.clear();
+  }
+
+ protected:
+  RfInputs<T> params;
+  RF_params rf_params;
+
+  T *data_d, *inference_data_d;
+  std::vector<T> data_h;
+  std::vector<T> inference_data_h;
+
+  int diff_elements = 0;
+  // Set to 1 for regression and 2 for binary classification
+  // #class for multi-classification
+  int task_category;
+  int is_classification;
+
+  int data_len;
+  int inference_data_len;
+
+  cudaStream_t stream;
+  cumlHandle handle;
+
+  std::vector<float> treelite_predicted_labels;
+  std::vector<float> ref_predicted_labels;
+
+  std::string test_dir;
+  std::string dir_name;
+
+  L *labels_d, *predicted_labels_d;
+  std::vector<L> labels_h;
+  std::vector<L> predicted_labels_h;
+
+  RandomForestMetaData<T, L>* forest;
+};
+
+template <typename T, typename L>
+class RfTreeliteTestClf : public RfTreeliteTestCommon<T, L> {
+ protected:
+  void testClassifier() {
+    this->test_dir = "./treelite_test_clf/";
+    this->is_classification = 1;
+    // task_category - 1 for regression, 2 for binary classification
+    // #class for multi-class classification
+    this->task_category = 2;
+
+    // Populate data (assume Col major)
+    this->data_h = {30.0, 1.0, 2.0, 0.0, 10.0, 20.0, 10.0, 40.0};
+    this->data_h.resize(this->data_len);
```

Comment:
`resize()` here (as well as for `labels_h` and `inference_data_h`) is not necessary, as the preceding assignment sets the correct size already.